messaging/src/main/java/org/axonframework/commandhandling/SimpleCommandHandlingComponent.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.commandhandling;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.QualifiedName;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * TODO This should be regarded as a playground object to verify the API. Feel free to remove, adjust, or replicate this class to your needs.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public class SimpleCommandHandlingComponent implements CommandHandlingComponent {
+
+    private final ConcurrentHashMap<QualifiedName, CommandHandler> commandHandlers;
```

Comment:
I'm wondering why is it ConcurrentHashMap? 
Is it possible / expected to register handler from different threads.
 I thought we're doing the registrations during configuration phase, and we're done.

Reply:
Likely not necessary, no. This was automated-Steven-mode making it a `ConcurrentHashMap` :sweat_smile:

Reply:
Actually, it is not impossible that command handlers are registered at runtime. There is nothing to prevent someone from dynamically registering command handlers. So we must assume some concurrency here.